pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -422,9 +422,14 @@ func (c *NPLController) handleAddUpdatePod(key string, obj interface{}) error {
 			podPorts[port] = struct{}{}
 			portData := c.portTable.GetEntryByPodIPPort(podIP, int(cport.ContainerPort))
 			if portData == nil { // rule does not exist
-				nodePort, err = c.portTable.AddRule(podIP, port)
-				if err != nil {
-					return fmt.Errorf("failed to add rule for Pod %s: %v", key, err)
+				if int(cport.HostPort) > 0 {
```

Comment:
Do you plan to support Pod spec change? Like hostPort is added/removed later after Pod creation?

Reply:
I don't think this can be updated for a running Pod:

> Pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or `spec.tolerations`. For `spec.tolerations`, you can only add new entries.

Reply:
Ok. Then it should be good.